pkg/tfbridge/replaceErrorProperties.go

```diff
@@ -0,0 +1,48 @@
+package tfbridge
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi-terraform-bridge/v3/unstable/metadata"
+)
+
+func replaceSubstring(s, old, new string) string {
+	return strings.ReplaceAll(s, old, new)
+}
+
+func ReplaceSubstrings(s string, replacements map[string]string) string {
+	for k, v := range replacements {
+		s = replaceSubstring(s, k, v)
+	}
+	return s
+}
+
+func getConfigReplacements(info ProviderInfo) (map[string]string, error) {
```

Comment:
It would be great if we could build this from `(map[string]*SchemaInfo, shim.SchemaMap)`. That would allow us to have the same error correction function run for the provider itself and for each resource.